src/adaXT/decision_tree/decision_tree.pyx

```diff
@@ -165,29 +179,192 @@ class DecisionTree(BaseModel):
+                        refit_objs.append(cur_node.right_child)
+                    elif isinstance(cur_node.right_child, refit_object):
+                        cur_node.right_child.add_idx(idx)
+
+                    cur_node = cur_node.right_child
+                depth += 1
+
+        all_idx = np.arange(0, indices.shape[0], dtype=np.int32)
+        leaf_builder = self.leaf_builder_class(X, Y, all_idx)
+        criteria = self.criteria_class(X, Y, sample_weight)
+        # Make refit objects into leaf_nodes
+        n_obs = len(refit_objs)
+        nodes = []
+        for i in range(n_obs):
+            obj = refit_objs[i]
+            leaf_indices = np.array(obj.indices, dtype=np.int32)
+            new_node = leaf_builder.build_leaf(
+                    i,
+                    leaf_indices,
+                    obj.depth,
+                    criteria.impurity(leaf_indices),
+                    leaf_indices.shape[0],
+                    obj.parent,
+                    )
+            new_node.visited = 1
+            nodes.append(new_node)
+            if obj.is_left:
+                obj.parent.left_child = new_node
+            else:
+                obj.parent.right_child = new_node
+        self.leaf_nodes = nodes
+
+    # Assumes that each node has been visited during __fit_new_leaf_nodes
+    def __squash_tree(self) -> None:
+
+        decision_queue = []
+        decision_queue.append(self.root)
+        while len(decision_queue) > 0:
+            cur_node = decision_queue.pop(0)
+            # If we don't have a decision node, just continue
+            if not isinstance(cur_node, DecisionNode):
+                continue
+
+            # If left child was not visited, then squash current node and right
+            # child
+            if (cur_node.left_child is None) or (cur_node.left_child.visited ==
+                                                 0):
+                parent = cur_node.parent
+                # Root node
+                if parent is None:
+                    self.root = cur_node.right_child
+                # if current node is left child
+                elif parent.left_child == cur_node:
+                    # update parent to point to the child that has been visited
+                    # instead
+                    parent.left_child = cur_node.right_child
+                else:
+                    parent.right_child = cur_node.right_child
+
+                cur_node.right_child.parent = parent
+
+                # Only add this squashed child to the queue
+                decision_queue.append(cur_node.right_child)
+
+            # Same for the right
+            elif (cur_node.right_child is None) or (cur_node.right_child.visited
+                                                    == 0):
+                parent = cur_node.parent
+                # Root node
+                if parent is None:
+                    self.root = cur_node.right_child
```

Comment:
Shouldn't this be the left child?

Reply:
Yes, this might be the error we previously saw.